src/test/java/streams/part2/exercise/Exercise4.java

```diff
@@ -0,0 +1,62 @@
+package streams.part2.exercise;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.junit.Test;
+
+public class Exercise4 {
+
+    /**
+     * Выбирает из текста наиболее часто встречающиеся слова.
+     * Подсчет слов выполняется без учета их регистра, т.е. "Привет", "привет", "пРиВеТ" - одно и то же слово.
+     * Если некоторые слова имеют одинаковую частоту, то в выходном списке они упорядочиваются в лексикографическом порядке.
+     * @param text Исходный текст в котором слова (в смешанном регистре) разделены пробелами.
+     * @param numberWords Количество наиболее часто встречающихся слов, которые необходимо отобрать.
+     * @return Список отобранных слов (в нижнем регистре).
+     */
+    private List<String> getFrequentlyOccurringWords(String text, int numberWords) {
+        return Pattern.compile("\\s+")
+                      .splitAsStream(text)
+                      .filter(Pattern.compile("[\\wа-яёА-ЯЁ]+").asPredicate())
```

Comment:
`\w` is including 0-9 and "_", so it's better to replace it with "a-zA-Z"

Reply:
Fix